### DISCUSSION TITLE
Why is there usage from ISR under Data Cache Reads?

### DISCUSSION THREAD

**Discussion by p1ckle-rick** [vercel/community#7178] [Help] [answered]
### Summary

Hello,

The build output does not have any static pages, there are only dynamic pages. But under the Usage tab on Vercel there are Data Cache Reads from Incremental Static Regeneration (ISR).

I have no idea where this is coming from because as far as I understand you must explicitly implement ISR which I didn't do.

I don't have any usage under Data Cache Writes, Data Cache Data Cache, Data Cache Bandwidth, and Data Cache Revalidations. But since I have usage under Data Cache Reads from ISR does this mean that some requests are trying to read from the Data Cache even though there is nothing in there? 

### Steps to Reproduce

<img width="819" alt="Screenshot 2024-06-12 at 3 22 23 PM" src="https://github.com/vercel/community/assets/13596285/213c9a56-d8fe-4c64-9bc5-59da2af46e1c">

**Comment by leerob** [comment]
Hey there. Is this code base open source? It is hard to tell exactly without access to the code. Alternatively, you could create a minimal reproduction for us to investigate.

**Reply by p1ckle-rick** [reply]
No, the code is not open source.

I am not sure how to create a minimal reproduction because I don't know what is causing this.

**Reply by p1ckle-rick** [reply]
What will you be looking for in the code?

Based on the build output all of my routes are dynamic, so why is ISR involved?

Is there anyway to find out the request that is executing the Data Cache Read?

**Reply by p1ckle-rick** [reply]
The only other thing that comes to mind...

https://vercel.com/docs/infrastructure/data-cache#comparing-with-isr-and-edge-cache
> When a page contains entirely static data, Vercel will use ISR to generate the whole page. However, when a page contains a mix of static and dynamic data, the dynamic data needs to be re-fetched when rendering the page. In this scenario, Vercel Data Cache is used to cache the static data to avoid slow origin fetches.

Although I am not exactly sure that is meant by `static and dynamic data` as well as why there are no Data Cache Writes. Maybe because the initial write (during deployment) to Data Cache is not shown 🤷

**Reply by bitttttten** [reply, marked as answer]
The Vercel Data Cache is working when you have `fetch` calls, it is trying to read from the vercel cache before doing your fetch request: https://nextjs.org/docs/app/building-your-application/data-fetching/fetching-caching-and-revalidating#fetching-data-on-the-server-with-fetch

[a comment by an automated account is not shown]
